src/components/Dropdown.tsx

```diff
@@ -31,37 +38,57 @@ const Dropdown = ({
       setSelected(selected[0] === value ? [] : [value]);
     }
   };
+  const maxOptionLength = 15;
   return (
-    <div className="text-xs cursor-pointer relative ">
+    <div
+      className="text-xs cursor-pointer relative "
+      tabIndex={0}
+      onBlur={() => setExpanded(false)}
+    >
       {/* Dropdown header/unexpanded display */}
       <div
-        className={`${selected.length === 0 ? "text-gray-400" : "text-black"} flex justify-between w-[150px] px-4 py-2 border border-slate-500 rounded-lg`}
-        onClick={() => setExpanded(!expanded)}
-        onBlur={() => setExpanded(false)}
+        className={`${selected.length === 0 ? "text-gray-400" : "text-black"} ${disabled ? "text-gray-400 bg-slate-200" : ""} flex justify-between items-center w-[150px] pl-4 pr-2 py-2 border border-slate-500 rounded-lg`}
+        onClick={() => !disabled && setExpanded(!expanded)}
       >
-        {selected.length === 0
-          ? "Select..."
-          : selected.length === 1
-            ? selected[0]
-            : `${selected.length} selected`}
+        <div className="text-nowrap overflow-hidden ">
+          {selected.length === 0
```

Comment:
NO COMPLEX TERNARIES

refactor this into function that does the thing that you're trying to do here and put it at the top of the file.

There's a bonus feature to functions which is that they have names. 

displaySelectedOptions(...)